### ISSUE TITLE
slurm sstat command, munge undefined symbol 

### ISSUE THREAD

**Issue by itzsimpl** [NVIDIA/deepops#720]
I have used deepops to deploy a slurm-cluster with a login node and two compute nodes. In the default `slurm.conf` created by deepops I have uncommented the line:
```
AccountingStorageEnforce=associations,limits,qos,safe
```
and setup an additional association where I added `GrpTRESMins=gres/gpu` and `MaxWall` limits.

All seems to work ok, except if I run the command `sstat -j <job.id>` where I get the following error:
```
sstat: symbol lookup error: /usr/local/lib/slurm/auth_munge.so: undefined symbol: slurm_get_log_level
```

Searching the internet leads me to suggestions of a linking order issue, but no actual solution;
- https://lists.schedmd.com/pipermail/slurm-users/2019-April/003253.html
- https://bugs.schedmd.com/show_bug.cgi?id=1508

any ideas?

**Comment by ajdecon** [comment]
I can confirm I see the same thing on a recent cluster, though interestingly it seems to only impact `sstat`. None of the other Slurm commands I've tested show this issue.

Based on the linked issues, I suspect this does have to do with some order-of-operations with libraries getting pulled in via `dlopen`.

I'll try to look at this some time this week, but no firm ETA as this kind of linking order issue can be really hard to pin down.

**Comment by mathrock74** [comment]
I'm seeing similar errors with a fresh install of Slurm with deepops 21.03 in Ubuntu 20.04 hosts.

example sstat output:

~# sstat -j 207
       JobID  MaxVMSize  MaxVMSizeNode  MaxVMSizeTask  AveVMSize     MaxRSS MaxRSSNode MaxRSSTask     AveRSS MaxPages MaxPagesNode   MaxPagesTask   AvePages     MinCPU MinCPUNode MinCPUTask     AveCPU   NTasks AveCPUFreq ReqCPUFreqMin ReqCPUFreqMax ReqCPUFreqGov ConsumedEnergy  MaxDiskRead MaxDiskReadNode MaxDiskReadTask  AveDiskRead MaxDiskWrite MaxDiskWriteNode MaxDiskWriteTask AveDiskWrite TRESUsageInAve TRESUsageInMax TRESUsageInMaxNode TRESUsageInMaxTask TRESUsageInMin TRESUsageInMinNode TRESUsageInMinTask TRESUsageInTot TRESUsageOutAve TRESUsageOutMax TRESUsageOutMaxNode TRESUsageOutMaxTask TRESUsageOutMin TRESUsageOutMinNode TRESUsageOutMinTask TRESUsageOutTot 
------------ ---------- -------------- -------------- ---------- ---------- ---------- ---------- ---------- -------- ------------ -------------- ---------- ---------- ---------- ---------- ---------- -------- ---------- ------------- ------------- ------------- -------------- ------------ --------------- --------------- ------------ ------------ ---------------- ---------------- ------------ -------------- -------------- ------------------ ------------------ -------------- ------------------ ------------------ -------------- --------------- --------------- ------------------- ------------------- --------------- ------------------- ------------------- --------------- 
sstat: error: plugin_load_from_file: dlopen(/usr/local/lib/slurm/auth_munge.so): /usr/local/lib/slurm/auth_munge.so: undefined symbol: slurm_conf
sstat: error: Couldn't load specified plugin name for auth/munge: Dlopen of plugin file failed
sstat: error: cannot create auth context for auth/munge
sstat: error: slurm_send_node_msg: g_slurm_auth_create: REQUEST_FED_INFO has authentication error: No such process
sstat: error: plugin_load_from_file: dlopen(/usr/local/lib/slurm/auth_munge.so): /usr/local/lib/slurm/auth_munge.so: undefined symbol: slurm_conf
sstat: error: Couldn't load specified plugin name for auth/munge: Dlopen of plugin file failed
sstat: error: cannot create auth context for auth/munge
sstat: error: slurm_send_node_msg: g_slurm_auth_create: REQUEST_JOB_STEP_INFO has authentication error: No such process
sstat: error: couldn't get steps for job 207

Job 207 is definitely running.
Since I'm very new to Slurm and never used sstat before I don't know if this relates to my other big issue that I never get any stats for RawUsage in sshare (which I reported to slurm-users mailing list today)

thanks for help

**Comment by Spartee** [comment]
I've been working on a wrapper for the slurm C API and ran into this same thing. I don't get this error running the C executable directly, but when I use the wrapper I got the same error as you.

The way I solved it was to set dlopen to ``RTLD_GLOBAL`` from python. see [this](https://docs.python.org/3/library/sys.html#sys.setdlopenflags)

e.g.
```python
import os, sys
sys.setdlopenflags(os.RTLD_NOW | os.RTLD_GLOBAL)
```

I realize this isn't sstat, but hopefully it's related enough to help. this was the only github issue/site I found with the exact same error message about slurm_conf instead of other symbols.
